src/main/java/seedu/address/logic/parser/SortCommandParser.java

```diff
@@ -0,0 +1,144 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.SortCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Person;
+
+/**
+ * Parses input arguments and creates a new SortCommand object
+ */
+public class SortCommandParser implements Parser<SortCommand> {
+    // maps the prefixes to their comparators
+    private static final HashMap<Prefix, Comparator<Person>> COMPARATOR_HASH_MAP = new HashMap<>() {{
+            put(Prefix.ADDRESS, Comparator.comparing(Person::getAddress));
+            put(Prefix.EMAIL, Comparator.comparing(Person::getEmail));
+            put(Prefix.GROUP_TAG, Comparator.comparing(Person::getGroupTags));
+            put(Prefix.MODULE_TAG, Comparator.comparing(Person::getModuleTags));
+            put(Prefix.NAME, Comparator.comparing(Person::getName));
+            put(Prefix.PHONE, Comparator.comparing(Person::getPhone));
+            put(Prefix.TELEGRAM_HANDLE, Comparator.comparing(Person::getTelegramHandle));
+        }};
+
+    // use this as the default sorting direction
+    private static final HashMap<Prefix, Boolean> IS_ASCENDING_DEFAULT_HASH_MAP = new HashMap<>() {{
+            put(Prefix.ADDRESS, true);
+            put(Prefix.EMAIL, true);
+            put(Prefix.GROUP_TAG, false);
+            put(Prefix.MODULE_TAG, false);
+            put(Prefix.NAME, true);
+            put(Prefix.PHONE, true);
+            put(Prefix.TELEGRAM_HANDLE, true);
```

Comment:
Text fields (address, email, name, phone, telegram) are sorted in ascending order by default.

Tag fields (group, module) are sorted in decreasing order by default. This is because we are likely to want persons with more common modules, and more relationships.

Reply:
LGTM! Remember to include this in the documentation.